test/testMPAppIDs.cpp

```diff
@@ -0,0 +1,37 @@
+#ifndef TESTMPAPPIDS_H
+#define TESTMPAPPIDS_H
+
+#include "pmpo_createTestMPMesh.hpp"
+#include "pmpo_defines.h"
+#include "pmpo_c.h"
+
+extern "C" {
+  void testAppIDPointer(MPMesh_ptr p_mpmesh);
+}
+
+void testAppIDPointer(MPMesh_ptr p_mpmesh) {
+  auto p_MPs = ((polyMPO::MPMesh*)p_mpmesh)->p_MPs;
+
+  int numAddedMPs = 2;
+  Kokkos::View<int*> added_mp2Elm_d("added_mp2Elm_d", numAddedMPs);
+  Kokkos::parallel_for("set addedMP2Elm", numAddedMPs, KOKKOS_LAMBDA (const int i) {
+    added_mp2Elm_d(i) = i;
```

Comment:
Please assert above that numElms is equal to or above numAddedMPs (otherwise mp2Elm can point to non-existent mesh elements).